src/redux/saga/employer.ts

```diff
@@ -1,19 +1,22 @@
-import { put, all, takeLatest, select } from 'redux-saga/effects';
-import axios from 'axios';
+import { put, takeLatest, select } from 'redux-saga/effects'
+import axios from 'axios'
+import { AppState } from '../types'
 
-import { registerEmployerSuccess } from '../../redux/actions/employer';
+import { registerEmployerSuccess } from '../../redux/actions/employer'
 
-function* registerEmployerSaga(credentials: Credential) {
+const credentials = (state: AppState) => state.employer.credentials
```

Comment:
Since you are not saving the `credential` payload in the store, you will always get an empty string. Fix the reducer first